### PR TITLE
Edit & expand text in SDK tutorials; reorganize chapter for better flow

### PR DESCRIPTION
**Pull request by chenette**
- Reorganize top-level sections and clarify header titles throughout
- Rewrite intro text for most sections
- Add prerequisites and overview of transaction families
- In SDK tutorials, add headers for Create, Take, Delete, and Addressing
- Add relevant snippets from the XO transaction family spec
  to the SDK tutorials

Signed-off-by: Anne Chenette <chenette@bitwise.io>

### CODE REVIEW

File: docs/source/app_developers_guide/overview.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+transaction families for its unique requirements.
+
+A transaction family includes these components:
+
+* A transaction processor to define the business logic for your application
+* A data model to record and store data
+* A client to handle the client logic for your application
+
+The transaction processor is responsible for registering with the validator,
+providing a transaction handler with an apply method and a transaction handler
```
Comment (dplumb94) [inline docs/source/app_developers_guide/overview.rst]: This paragraph could be revised, not quite sure what "providing a transaction handler with an apply method and a transaction handler class..." means

File: docs/source/app_developers_guide/overview.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+class for metadata, and getting/setting state as needed.
+
+The client is responsible for creating and signing transactions, combining
+those transactions into batches, and submitting them to the validator. The
+client can post batches through the REST API or connect directly to the
+validator via 0MQ (also called ZMQ).
```
Comment (dplumb94) [inline docs/source/app_developers_guide/overview.rst]: Could just say ZeroMQ and possibly link to their repo

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 3]
```diff
@@ -12,55 +12,18 @@
...
-***********************************************
-Transaction Processor Tutorial  {{ language }}
-***********************************************
+*****************************************************
+Transaction Processor: Creating a Transaction Handler
+*****************************************************
 
-Overview
-========
+A transaction processor has two top-level components: a processor class and
```
Comment (boydjohnson) [inline docs/source/_templates/sdk_TP_tutorial.rst]: With calling out the processor class, I would mention that the processor class is completely supplied by the sdk, whereas the handler class is written by the transaction family writer.

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 4]
```diff
@@ -1150,8 +1156,29 @@ into bytes that can be stored in the validator's Radix-Merkle tree.
...
+first 64 characters of the SHA-512 hash of the UTF-8 encoding of the
+game name.
+
+For example, the XO address for a game called "mygame" could be
+generated as follows:
+
+.. code-block:: pycon
+
+    >>> hashlib.sha512('xo'.encode('utf-8')).hexdigest()[:6] +
+    >>> hashlib.sha512('mygame'.encode('utf-8')).hexdigest()[:64]
```
Comment (dcmiddle) [inline docs/source/_templates/sdk_TP_tutorial.rst]: nit, but if you make other changes, you use "my-game" in #1466 
If you update that here your hash will be ... 
```>>> x = hashlib.sha512('xo'.encode('utf-8')).hexdigest()[:6]
>>> x
'5b7349'
>>> y = hashlib.sha512('my-game'.encode('utf-8')).hexdigest()[:64]
>>> y
'4d4cffe9cf3fb4e41def5114a323e292af9b0e07925cca6299d671ce7fc7ec37'
>>> x+y
'5b73494d4cffe9cf3fb4e41def5114a323e292af9b0e07925cca6299d671ce7fc7ec37'
```

File: docs/source/_templates/sdk_overview_tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+distributed version of the two-player game
+`tic-tac-toe <https://en.wikipedia.org/wiki/Tic-tac-toe>`_.
+
+This tutorial also describes how a client can use the {{ language }} SDK
+to create transactions and submit them as :term:`Sawtooth batches<Batch>`.
+
+.. note::
+
+   This tutorial demonstrates the relevant concepts for a Sawtooth transaction
+   processor and client, but does not  create a complete implementation.
```
Comment (dcmiddle) [inline docs/source/_templates/sdk_overview_tutorial.rst]: double space between `not create`

File: docs/source/app_developers_guide.rst [hunk 6]
```diff
@@ -3,27 +3,32 @@
...
-the use of Sawtooth's provided SDKs and :term:`REST API`. Topics include
-development of :term:`transaction families<Transaction family>` and associated
-client programs. Transaction families codify business rules used to modify
-state, while client programs typically submit transactions and view state.
+the use of Sawtooth's provided SDKs and :term:`REST API`.
 
-SDKs are provided in several languages: C++, Go, Java, Javascript, and Python.
+Topics include developing a :term:`transaction family<Transaction family>` and
+an associated client program. Transaction families codify business rules used
+to modify state, while client programs typically submit transactions and view
```
Comment (dcmiddle) [inline docs/source/app_developers_guide.rst]: If we are going to use TF as the umbrella over client, server, and data model then this description should be updated. It appears to be more of a string replace on TP

Comment (dcmiddle) [reply docs/source/app_developers_guide.rst]: If we are going to use TF as the umbrella over client, server, and data model then this description should be updated. It appears to be more of a string replace on TP

File: docs/source/_templates/sdk_TP_tutorial.rst (hunk 4 above)
Comment (chenette) [reply docs/source/_templates/sdk_TP_tutorial.rst]: THANKS!!!!

**Pull request merged**